packages/roosterjs-editor-plugins/lib/Paste/wacConverter/convertPastedContentFromWac.ts

```diff
@@ -0,0 +1,128 @@
+import { HtmlSanitizer } from 'roosterjs-html-sanitizer';
+
+const WORD_ONLINE_IDENTIFYING_SELECTOR = "div.ListContainerWrapper,div.OutlineElement";
+
+const LIST_CONTAINER_ELEMENT_CLASS_NAME = "ListContainerWrapper";
+
+const UNORDERED_LIST_TAG_NAME = "UL";
+
+const ORDERED_LIST_TAG_NAME = "OL";
+
+export function isWAC(node: HTMLElement): boolean {
+    if (node) {
+        return node.querySelector(WORD_ONLINE_IDENTIFYING_SELECTOR) ? true : false;
+    }
+    return false;
+}
+
+/**
+ * Convert text copied from word online into text that's workable with rooster editor
+ * @param doc Document that is being pasted into editor.
+ */
+export default function convertPastedContentFromWac(doc: HTMLDocument) {
+    // From word online rows are wrapped with OutlineGroup class, which
+    // makes each row not direct siblings. For the ease of converting,
+    // the following code flatens the row elements and make them direct
+    // siblings.
+    const wrapperElement = document.createElement('div');
+    let el = doc.querySelector(WORD_ONLINE_IDENTIFYING_SELECTOR);
+    while (el) {
+        el.parentElement.removeChild(el);
+        wrapperElement.appendChild(el);
+        el = doc.querySelector(WORD_ONLINE_IDENTIFYING_SELECTOR);
+    }
+
+    // In word online list items are wrapped in a container
+    // and adjacent list items are belong to the same list.
+    // In the following comments we will call a block of adjacent
+    // list items a list item block.
+
+    // Get the first element of a list item block.
+    let listContainerElement = wrapperElement.getElementsByClassName(LIST_CONTAINER_ELEMENT_CLASS_NAME)[0];
+    while (listContainerElement) {
+        let convertedListElement: Element; // Element that store converted list item in current list item block.
+        let currentListContainer = listContainerElement; // Iterator that iterate through each list item in the current block.
+        let listContainerParent = currentListContainer.parentElement; // Parent element of current list item block.
+        // if the iterator exists and the iterator still contains list then perceed with loop.
+        while (currentListContainer && currentListContainer.classList.contains(LIST_CONTAINER_ELEMENT_CLASS_NAME)) {
+            let listType: string; // list type that is contained by iterator.
+
+            if (currentListContainer.querySelector(UNORDERED_LIST_TAG_NAME)) {
```

Comment:
Does this mean if 'UL' class applied to any of current node's child node, we will use UL? Or UL class must be applied to current node? I think we need a more strict check